### PR TITLE
Add support for DropMenu.Option onClick handler that is passed through

### PR DESCRIPTION
**Pull request by ogupte**
## PR Checklist

- Manually tested across supported browsers
  - [ ] Chrome
  - [ ] Firefox
  - [ ] Safari
  - [ ] IE11 (Win 7)
  - [ ] Edge (Win 10)
- [ ] Unit tests written (`common` at minimum)
- [ ] PR has one of the `semver-` labels
- [ ] Two core team engineer approvals
- [ ] One core team UX approval

### CODE REVIEW

File: src/components/DropMenu/DropMenu.jsx [hunk 1]
```diff
@@ -414,6 +414,7 @@ const DropMenu = createClass({
 
 		if (!optionProps.isDisabled && focusedIndex !== optionIndex) {
 			onFocusOption(optionIndex, { props: optionProps, event });
+			optionProps.onClick({{ props: optionProps, event }});
```
Comment (jondlm) [inline src/components/DropMenu/DropMenu.jsx]: Does this break when there is no `onClick`?

Comment (chrisdopuch) [inline src/components/DropMenu/DropMenu.jsx]: ➕ 1 you could add a `_.noop` to the `getDefaultProps`

Comment (mute) [inline src/components/DropMenu/DropMenu.jsx]: It seems weird to fire two events for a single action; should it be a precedence thing? e.g.:
`if optionProps.onClick is set` -> call that function
`else -> ` call `onSelect` function

**Comment by ogupte** [comment]
this PR was for illustration and discussion purposes, closing it now.

**Pull request closed without merging**